In [1]:
import pandas as pd
import warnings as wr
wr.filterwarnings('ignore')

In [4]:
df = pd.read_csv('weight-height.csv')
df.head()

Gender     Height      Weight
0   Male  73.847017  241.893563
1   Male  68.781904  162.310473
2   Male  74.110105  212.740856
3   Male  71.730978  220.042470
4   Male  69.881796  206.349801

In [5]:
df.shape

(8555, 3)

In [6]:
from sklearn.preprocessing import LabelEncoder
le = LabelEncoder()

In [7]:
df.Gender = le.fit_transform(df[['Gender']])

In [8]:
df.head()

Gender     Height      Weight
0       1  73.847017  241.893563
1       1  68.781904  162.310473
2       1  74.110105  212.740856
3       1  71.730978  220.042470
4       1  69.881796  206.349801

In [9]:
df.tail()

Gender     Height      Weight
8550       0  60.483946  110.565497
8551       0  63.423372  129.921671
8552       0  65.584057  155.942671
8553       0  67.429971  151.678405
8554       0  60.921791  131.253738

# Go Ahead

In [10]:
X = df[['Gender', 'Height']]
X.head()

Gender     Height
0       1  73.847017
1       1  68.781904
2       1  74.110105
3       1  71.730978
4       1  69.881796

In [11]:
y = df['Weight']
y.head()

0    241.893563
1    162.310473
2    212.740856
3    220.042470
4    206.349801
Name: Weight, dtype: float64

In [13]:
from sklearn.model_selection import train_test_split

In [14]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

In [15]:
len(X_train)

5988

<h3>Apply Linear Regression</h3>

In [19]:
from sklearn.linear_model import LinearRegression

In [20]:
model1 = LinearRegression()
model1.fit(X_train, y_train)

LinearRegression()

<h4>Training Accuracy</h4>

In [22]:
model1.score(X_train, y_train)

0.8973793060969246

<h4>Testing Accuracy</h4>

In [23]:
model1.score(X_test, y_test)

0.905911242442266

<h4>Mean Squared Error for Testing</h4>

In [24]:
from sklearn.metrics import mean_squared_error

In [29]:
y_pred1 = model1.predict(X_test)
y_pred1[:20]

array([142.44868114, 186.0517281 , 198.42283706, 129.88427401,
       190.17143626, 125.69023003, 176.96797911, 173.97033314,
       203.15659046, 119.35796412, 191.05790038, 148.14570554,
       215.75859345, 177.17487427, 218.34996229, 185.48491883,
       211.30556429, 168.28262279, 142.44345794, 197.15660922])

In [30]:
mse1 = mean_squared_error(y_test, y_pred1)
mse1

96.83734437830608

<h3>Apply K-Neighbors Regressor</h3>

In [31]:
from sklearn.neighbors import KNeighborsRegressor

In [45]:
neigh = KNeighborsRegressor(weights='uniform')

In [46]:
neigh.fit(X_train, y_train)

KNeighborsRegressor()

<h4>Training Accuracy</h4>

In [47]:
neigh.score(X_train, y_train)

0.9172142723737918

<h4>Testing Accuracy</h4>

In [48]:
neigh.score(X_test, y_test)

0.8821086423439015

In [49]:
y_pred2 = neigh.predict(X_test)
y_pred2[:20]

array([142.1445657 , 181.38276924, 187.07075266, 113.00549496,
       186.65895774, 119.337843  , 170.92474752, 174.5632046 ,
       207.68255074, 121.45720974, 197.70694576, 144.96461618,
       217.70515066, 172.48179168, 218.35646112, 183.34871712,
       210.93764908, 169.6801828 , 141.62367836, 199.39569894])

<h4>Mean Squared Error for Testing</h4>

In [50]:
mse2 = mean_squared_error(y_test, y_pred2)
mse2

121.33528273624482

<h3 style='color:green'>Here we saw that, When we apply Linear Regression throughout this dataset we got 89% for training accuracy and 90% for testing accuracy. On the other hand When we apply KNeighborsRegressors we got 91% for training accuracy and 88% for testing accuracy. We can compare another thing is that, In Linear Regression Mean Squared Error having value of 96.84 where in KNeighborsRegressor we got 121.34. So we can conclude that for Linear Regression we will get less error than KNN</h3>